test/AutoDiff/differentiating_attr_type_checking.swift

```diff
@@ -280,3 +280,16 @@ extension InstanceMethodProto where Self : Differentiable {
     return (bar(), { _ in .zero })
   }
 }
+
+// Test consistent usages of `@differentiable` and `@differentiating` where
+// derivative functions are specified in both attributes.
+@differentiable(jvp: jvpConsistent, vjp: vjpConsistent)
```

Comment:
Sorry, I forgot to add "both attributes".

The test is fixed. It basically tests this scenario:
```swift
@differentiable(vjp: vjpFoo)
func foo(_ x: Float) -> Float {
  return x
}
// Previously, type-checking `@differentiating` below produced an error, because
// `foo` already has a `@differentiable` attribute with the same `wrt` parameters.
// Now, an error is emitted only if the `@differentiable` attribute has a different
// registered derivative function (not the `@differentiating` function).
@differentiating(foo)
func vjpFoo(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
  return (x, { $0 })
}
```